include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -1016,7 +1019,7 @@ class open_addressing_ref_impl {
                            InputProbeIt input_probe_end,
                            OutputProbeIt output_probe,
                            OutputMatchIt output_match,
-                           AtomicCounter& atomic_counter) const
+                           AtomicCounter* atomic_counter) const
```

Comment:
Using a reference is fine but not ideal since the counter is passed as a pointer to the atomic through `counter_storage::data()`, which returns a pointer. Typically, we avoid passing a pointer by reference, for example, we rarely do:
```cpp
void func (int *& counter) { ... }
```

Reply:
I see. Since it's in detail namespace this is absolutely valid.